invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -651,17 +653,8 @@ private InvoiceWithFutureNotifications processAccountWithLockAndInputTargetDate(
                 // Commit invoice on disk
                 final ExistingInvoiceMetadata existingInvoiceMetadata = new ExistingInvoiceMetadata(existingInvoices);
                 commitInvoiceAndSetFutureNotifications(account, invoiceModelDao, billingEvents, trackingIds, futureAccountNotifications, existingInvoiceMetadata, internalCallContext);
+                setChargedThroughDatesNoExceptions(invoiceWithMetadata.getChargeThroughDates(), internalCallContext);
```

Comment:
Should we really ignore the exception here?

Reply:
There is no good solution. My 2021 view is, that if we come from api, we should throw, but if we come from bus event, we should really park the account as the state becomes inconsistent (but this is not what we do prior my change I think, so as far as I can tell, I tied to current behavior). Let's talk?